samples/cellular/modem_shell/src/location/location_shell.c

```diff
@@ -325,10 +366,39 @@ void location_ctrl_event_handler(const struct location_event_data *event_data)
 #endif
 #endif /* defined(CONFIG_LOCATION_SERVICE_EXTERNAL) */
 	case LOCATION_EVT_RESULT_UNKNOWN:
-		mosh_print("Location request completed, but the result is not known");
+		mosh_print("Location request completed, but the result is not known:");
+		mosh_print(
+			"  used method: %s (%d)",
+			location_method_str(event_data->method),
+			event_data->method);
 		break;
 	case LOCATION_EVT_STARTED:
-		mosh_print("Location request has been started");
+		mosh_print("Location request has been started:");
+		mosh_print(
+			"  used method: %s (%d)",
+			location_method_str(event_data->method),
+			event_data->method);
+		break;
+	case LOCATION_EVT_FALLBACK:
+		mosh_print("Location request fallback has occurred:");
+		mosh_print(
+			"  failed method: %s (%d)",
+			location_method_str(event_data->method),
+			event_data->method);
+		mosh_print(
+			"  new method: %s (%d)",
+			location_method_str(event_data->fallback.next_method),
+			event_data->fallback.next_method);
+		mosh_print(
+			"  cause: %s",
+			(event_data->fallback.cause == LOCATION_EVT_TIMEOUT) ?
+			 "timeout" :
+			 (event_data->fallback.cause == LOCATION_EVT_ERROR) ?
+			  "error" :
+			  "unknown");
+#if defined(CONFIG_LOCATION_DATA_DETAILS)
```

Comment:
`LOCATION_EVT_STARTED` and `LOCATION_EVT_FALLBACK` events can only be received when `CONFIG_LOCATION_DATA_DETAILS` is enabled. Cases for those events could be flagged to save memory when data details are not enabled. If the events would be unexpectedly received, a warning would be printed anyway.